### DISCUSSION TITLE
P/ Dependency Management Process

### DISCUSSION THREAD

**Discussion by 6senseHQ** [6sense-Technologies/6senseHQ-Discussions#11] [General Process]
#### 1. **Updating the Project Tracking Sheet**
   - **New Columns Added**:
     - **ENV Dependency**: Tracks environment variable dependencies.
     - **Client Dependency**: Tracks tasks that require client action or input.
     - **3rd Party Dependency**: Tracks dependencies involving third-party services or APIs.

#### 2. **Dependency Identification**
   - **Developer Notification**:
     - After completing an R&D task, the developer identifies any dependencies.
     - The developer notifies the Project Manager (PM) via a GitHub Discussion.
   - **PM Action**:
     - The PM updates the relevant columns in the project tracking sheet.
   - **Instructional Materials**:
     - Each dependency must be documented with an instructional link or video.
     - **For Clients/PM**:
       - A list of dependencies and steps to resolve them.
     - **For SQA**:
       - ENV Dependency names and Third Party Dependency names.
       - Instructions on retrieving and verifying values.

#### 3. **Verification Process**
   - **ENV Dependency Verification**:
     1. **Application Verification**:
        - A task is created for Kubernetes deployment.
        - The task includes a screenshot showing the success message and timestamp.
     2. **Value Verification**:
        - The tester checks the presence of the required property in AWS.
        - Values are verified for accuracy based on developer-provided instructions.
   - **3rd Party Dependency Verification**:
     - Follow the provided instructions to ensure correct integration with third-party services.
   - **Client Dependency Verification**:
     1. **Task Assignment**:
        - The PM assigns a verification task to the client.
     2. **Client Action**:
        - The client completes the task and provides a screenshot as proof.
     3. **PM Review**:
        - The PM moves the task to the “Preparing for QC” column and assigns it to a responsible person.
     4. **Final Verification**:
        - The assigned person verifies the dependency, adds comments, and sends the task to testing.

#### 4. **Frontend-Backend Coordination**
   - **Cross-team Communication**:
     - If the frontend team identifies backend R&D requirements, they notify the PM.
     - The PM assigns the task to the backend team, and the process works vice versa.

#### 5. **Client Confirmation**
   - **Post-SQA Approval**:
     - Once the SQA team approves the tasks, the PM confirms completion with the client.

#### 6. **Handling Changes**
   - **Change Verification**:
     - The verification process remains unchanged for any value changes.
     - The developer creates a new task for testing when a value is modified.
     - SQA re-verifies all steps associated with the change.

#### 7. **Training Materials**
   - **Instructional Content**:
     - A video guide on checking ENV values in the AWS Parameter Store needs to be created and added to the training resources. And it should be created before the testing period.

CC: @ahsanaasim

**Comment by 6senseHQ** [comment]
Meeting docs [link](https://docs.google.com/document/d/16qYyg1cN0B8UWnUCg29v_RxUefTb3sP3TrUEQsVup-g/edit#heading=h.mi4your03uk3)

By farjana

**Comment by talha6sensehq** [comment]
![image](https://github.com/user-attachments/assets/5821767d-9aec-4a58-afe3-e8793320eb5b)

Who is the PM that is mentioned?

**Reply by 6senseHQ** [reply]
Everything said to notify PM is about Farjana Ahmed here. @talha6sensehq 

By Farjana
